### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by dr-bizz**
## Description
In this PR, I update dependencies which have been flagged by the Dependabot app.

**Major Upgrades:**
angular-eslint packages -> 17.5.0
zone.js -> 0.15.0 - I had to make some slight adjustments to the test.ts file due to changes in the package.

**Patch Upgrades:**
eslint-plugin-prettier
karma-jasmine-html-reporter
prettier
pretty-quick

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,16 +53,16 @@
...
     "husky": "^8.0.0",
     "jasmine-core": "~5.4.0",
     "karma": "^6.4.4",
     "karma-chrome-launcher": "^3.2.0",
     "karma-coverage-istanbul-reporter": "^3.0.3",
     "karma-jasmine": "^5.1.0",
-    "karma-jasmine-html-reporter": "^1.7.0",
-    "prettier": "^3.0.0",
-    "pretty-quick": "^3.1.0",
+    "karma-jasmine-html-reporter": "~2.1.0",
```
Comment (canac) [inline package.json:63]: Is there a reason to switch these from `^` to `~`?

**Review by canac — APPROVED** [review]
Looks good!

**Review by aaronlaib — APPROVED** [review]
Looks good!

File: package.json (hunk 1 above)
Comment (dr-bizz) [reply package.json:63]: No, I think I got them confused.

**Comment by canac** [comment]
@dr-bizz Make sure to run `yarn` again and commit the changes after your changes to `package.json`.

**Pull request merged**